cal/cal/helpers.py

```diff
@@ -28,6 +31,43 @@ def ensure_timezone_awareness(dt, optional_timezone=None):
     return dt
 
 
+def handle_time_string(time_str, timezone_str):
+    """
+    Converts UTC 'time_str' to timezone represented by 'timezone_str'
```

Comment:
This description need to be improved. If you look closely at the logic, the timezone is first ignored if a timezone could be parsed from the time string. Then the timezone is applied to the time so that it is both timezone aware and has a timezone offset. It is not always converted, since we don't want to convert datetimes that are already timezone aware.